components/sigils.js

```diff
@@ -0,0 +1,39 @@
+import React, { useState, useEffect } from "react";
+import Select from "react-select";
+
+const Sigils = (props) => {
+  const [sigil, setSigil] = useState("");
+  const { setSigilsTF } = props;
+
+  const options = [
+    { value: "", label: "No sigils" },
+    { value: "v1644605839:Inscryption:Sigils:airborne", label: "Airborne" },
+    { value: "v1644605839:Inscryption:Sigils:bifurcated_strike", label: "Bifurcated Strike" },
+  ];
+
+  useEffect(() => {
+    sigil === ""
+      ? setSigilsTF("")
+      : setSigilsTF(`l_${sigil}.png,g_south,y_64/`);
+  }, [sigil]);

```

Comment:
Is there any benefit to doubling up on setting states? Here you call `setSigil` and then based on that result `setSigilsTF` is triggered with that result. Could you not entirely handle this by passing an `onChange` into this component from the base page, or just calling `setSigilsTF` as part of the `onChange` below? Not a criticism, just unsure about if theres a benefit to doing it this way.

Reply:
Yep, it's absolutely possible to do either tbh. My only justification is in other components (see [name.js](https://github.com/delete-44/inscryber/blob/0135286515a121c59d2389e4b01d870cac98bb8a/components/name.js#L11)) the transformation that needs to be set is a lot more complex & doing that all inline becomes less readable. I'd rather keep the behaviour of the components consistent instead of having this one handle things differently because it's a simpler transformation

Reply:
Could you achieve the same with a helper function like:

```
  setSigil(value) {
    value === ""
      ? setSigilsTF("")
      : setSigilsTF(`l_${value}.png,g_south,y_64/`);
    }
```

Genuine question, is there a functional difference between just defining a function above and calling it in the below onChange? My understanding of react is that you're currently setting the state in the component, waiting for a re-render and then triggering a second state change in the parent page when the useEffect triggers, when defining a function could likely skip the initial re-render step. I'm perhaps thinking of this in terms of Flutter more than React, however.

Reply:
There is a fundamental difference here, which is that the input field is `controlled`, which means that it relies on me updating the state for it to rerender. IE if I don't call `setSigils`, the user will never see the view update. 

I need to control two bits of state - `sigils` and `sigilsTF`. So if I made it a function, it would be:
```
  changeCallback(value) {
    setSigil(value);
    value === ""
      ? setSigilsTF("")
      : setSigilsTF(`l_${value}.png,g_south,y_64/`);
    }
```

which then causes the rerender anyway. The code as it currently stands basically does this with one additional function call in between, so the savings would be minimal for sacrificing the usefulness of being able to work with the current live version of "sigils" at any given point

Reply:
Fair enough. That's all good to me then. Like I said, I'm used to Flutter more than React. A lot of the input fields I work with tend to handle their own state internally and therefore don't need a re-render to be manually triggered. I genuinely wasn't aware.